lib/ulp.c

```diff
@@ -870,42 +878,44 @@ compare_build_ids(struct dl_phdr_info *info,
 
     build_id_len = (uint32_t) * (note_ptr + 4);
     build_id_len += build_id_len % 4;
-    if (build_id_len != ulp->objs->build_id_len)
-      return 0;
 
     /* we compute, but currently do not check note names */
     name_len = (uint32_t)*note_ptr;
     name_len += name_len % 4;
 
     build_id_ptr = note_ptr + 12 + name_len;
-    if (memcmp(ulp->objs->build_id, build_id_ptr, build_id_len) == 0) {
-      ulp->objs->build_id_check = 1;
-      return 0;
-    }
-    else {
-      return 1;
```

Comment:
After this change, the loop in `dl_iterate_phdr` will no longer stop after a build-id mismatch (the loop will run until all objects in the process have been examined). That's OK, because live patch application is not in the critical path (and also because mismatches are probably rare).

Reply:
I did wonder about this but kept it safe for testing. I also wondered if we can stop the loop after a miss match could we also stop it after a match or would that return an error code?

Reply:
> I also wondered if we can stop the loop after a miss match could we also stop it after a match or would that return an error code?

I think we can, yes. We don't use the return value of dl_iterate_phdr anywhere.

That said, I'm fine with stopping after a match but also fine with not stopping at all.